components/style/selector_matching.rs

```diff
@@ -609,16 +609,16 @@ impl PerPseudoElementSelectorMap {
 #[cfg_attr(feature = "servo", derive(HeapSizeOf))]
 pub struct SelectorMap {
     // TODO: Tune the initial capacity of the HashMap
-    id_hash: FnvHashMap<Atom, Vec<Rule>>,
-    class_hash: FnvHashMap<Atom, Vec<Rule>>,
-    local_name_hash: FnvHashMap<Atom, Vec<Rule>>,
+    pub id_hash: FnvHashMap<Atom, Vec<Rule>>,
```

Comment:
Yeah, this is unfortunate but the same has happened in many other cases. As long as tests are not in the same crate, things used by tests have to be public. On the other hand having in-crate unit tests doubles the compile time for that crate. (You have to build everything again with `--test`.)